nifi-nar-bundles/nifi-adx-bundle/nifi-adx/src/main/java/org/apache/nifi/adx/AzureAdxSourceConnectionService.java

```diff
@@ -0,0 +1,186 @@
+            .build();
+
+    public static final PropertyDescriptor APP_ID = new PropertyDescriptor
+            .Builder().name(AzureAdxConnectionServiceParameter.APP_ID.name())
+            .displayName(AzureAdxConnectionServiceParameter.APP_ID.getParamDisplayName())
+            .description(AzureAdxConnectionServiceParameter.APP_ID.getDescription())
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor APP_KEY = new PropertyDescriptor
+            .Builder().name(AzureAdxConnectionServiceParameter.APP_KEY.name())
+            .displayName(AzureAdxConnectionServiceParameter.APP_KEY.getParamDisplayName())
+            .description(AzureAdxConnectionServiceParameter.APP_KEY.getDescription())
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor APP_TENANT = new PropertyDescriptor
+            .Builder().name(AzureAdxConnectionServiceParameter.APP_TENANT.name())
+            .displayName(AzureAdxConnectionServiceParameter.APP_TENANT.getParamDisplayName())
+            .description(AzureAdxConnectionServiceParameter.APP_TENANT.getDescription())
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor CLUSTER_URL = new PropertyDescriptor
+            .Builder().name(AzureAdxConnectionServiceParameter.CLUSTER_URL.name())
+            .displayName(AzureAdxConnectionServiceParameter.CLUSTER_URL.getParamDisplayName())
+            .description(AzureAdxConnectionServiceParameter.CLUSTER_URL.getDescription())
+            .required(true)
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = List.of(KUSTO_AUTH_STRATEGY,APP_ID,APP_KEY,APP_TENANT,CLUSTER_URL);
+
+    private Client executionClient;
+
+    private ADXConnectionParams adxConnectionParams;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    /**
+     * @param context the configuration context
+     */
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) throws ProcessException {
+        getLogger().info("Starting Azure ADX Source Connection Service...");
+        adxConnectionParams = new ADXConnectionParams();
+        adxConnectionParams.setKustoAuthStrategy(context.getProperty(KUSTO_AUTH_STRATEGY).evaluateAttributeExpressions().getValue());
+        adxConnectionParams.setAppId(context.getProperty(APP_ID).evaluateAttributeExpressions().getValue());
+        adxConnectionParams.setAppKey(context.getProperty(APP_KEY).evaluateAttributeExpressions().getValue());
```

Comment:
Sensitive properties should not support attribute evaluation.
```suggestion
        adxConnectionParams.setAppKey(context.getProperty(APP_KEY).getValue());
```